### PR TITLE
Revert "Fix `--sync` missing some recipes"

### PR DESCRIPTION
**Pull request by kylewlacy**
Reverts brioche-dev/brioche#122

Okay, it finally hit me what was happening...

So the way it was previously implemented `--sync` would find all top-level recipes for a build, then would sync them plus any recipes that they referenced _by hash_. #122 changed it so it would find _all_ internal recipes, either by hash or nested directly. The PR changed it so these internal recipes would be synced too

I think it would be a valid decision to sync these internal recipes, but for the purposes of `--sync`, I think it's the wrong choice. Namely, it just pollutes the registry with some recipes that should never get directly downloaded (clients only download the recipes referenced by hash, they would never have a need to download a recipe embedded directly)

As for the issue with `eza`, I finally realized that it was just because some of the bakes I had locally were non canonical (meaning that what was stored in my local DB was different than the registry), which meant that later recipes diverged enough that I got a cache miss.

Basically, all that is to say is that everything was already working as expected prior to #112, although the end result was unintuitive. Upon reflection, I felt that the earlier behavior was more in line with how syncing should work

### CODE REVIEW

**Comment by kylewlacy** [comment]
As part of #112, I ended up re-running the latest CI build for `brioche-packages` ([CI #278](https://github.com/brioche-dev/brioche-packages/actions/runs/10717862543)). This meant the extra recipes uploaded from #112 already ended up in the registry

As part of the revert, I rolled back the database to 2024-09-07 at 12:00am. Since the `brioche-packages` repo hadn't been updated in a while, this only impacted the newly inserted recipes, so the effects of #112 have been undone in the registry too
